### PR TITLE
Add embroider compatibility for YUIDoc

### PR DESCRIPTION
**Pull request by dbendaou**
In Embroider world, there's no `postprocessTree` hook. 

Therefore, the way we generate YUIDoc [here](https://github.com/storybookjs/ember-cli-storybook/blob/master/index.js#L32) won't work.

This refacts the index.js to be able to process YUIDoc still while being compatible with Embroider. 

Heavily inspired by https://github.com/ef4/prember/pull/64/files

Resolves https://github.com/storybookjs/ember-cli-storybook/issues/153

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,8 +81,19 @@ So in order to add a script tag to the generated `preview-head.html` a potential
...
+
+## Embroider & YUIDoc
+If you want to use YUIDoc with Embroider, you will need to modify your `ember-cli-build.js` to be able to compile the documentation.
+
+```diff
+const { Webpack } = require('@embroider/webpack');
+- return require('@embroider/compat').compatBuild(app, Webpack);
++ const compiledApp = require('@embroider/compat').compatBuild(app, Webpack);
++
++ return require('prember').prerender(app, compiledApp);
```
Comment (Turbo87) [inline README.md:94]: `require('prember')` ? 🤔

File: index.js [hunk 2]
```diff
@@ -27,40 +26,70 @@ module.exports = {
...
+  prerender(app, tree) {
+    let storybookAddon = app.project.addons.find(
+      ({ name }) => name === '@storybook/ember-cli-storybook'
+    );
+
+    if (!storybookAddon) {
+      throw new Error(
+        "Could not find initialized ember-cli-storybook addon. It must be part of your app's dependencies!"
+      );
+    }
```
Comment (Turbo87) [inline index.js:53]: can't we just use `this`? 🤔

File: index.js [hunk 3]
```diff
@@ -27,40 +26,70 @@ module.exports = {
...
-    }
-
     let componentJS = new Funnel('.', {
       include: componentFilePathPatterns,
     });
 
-    if (!YUIDocsGenerator) {
-      YUIDocsGenerator = require('ember-cli-addon-docs-yuidoc/lib/broccoli/generator');
-    }
-
```
Comment (Turbo87) [inline index.js:53]: why did you remove this?

File: README.md (hunk 1 above)
Comment (dbendaou) [reply README.md]: Ooops, should be `@storybook/ember-cli-storybook`. Updated!

File: index.js (hunk 3 above)
Comment (dbendaou) [reply index.js]: Unintentional, my bad 😄

File: index.js (hunk 2 above)
Comment (dbendaou) [reply index.js:53]: I've taken this example from what @ef4 did on this MR https://github.com/ef4/prember/pull/64/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R22 

using `this` should work but I would rather stay as close as what he did since I'm still not 100% comfortable with Embroider yet and more specifically what it does under the hood and potential edge-case

**Comment by gossi** [comment]
What about adding embroider as test-scenario for ember-try?

**Comment by dbendaou** [comment]
I've open a dedicated PR here https://github.com/storybookjs/ember-cli-storybook/pull/161

**Comment by gossi** [comment]
Oka, I approved the other PR, please merge and rebase this one, so CI has a chance to actually test your changes here :)

**Comment by dbendaou** [comment]
All good, all green 🟢

**Review by gossi — APPROVED** [review]
